src/Maestro/tests/Maestro.ScenarioTests/ScenarioTests_GitHubFlow.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.DotNet.Maestro.Client.Models;
+using NUnit.Framework;
+
+namespace Maestro.ScenarioTests
+{
+    [TestFixture]
+    [Category("PostDeployment")]
```

Comment:
This should probably include `[NonParallelizable]` on it, since it's got multiple tests and a bunch of state.

Reply:
So Ricardo & I actually came up with a solution for that - every test runs in its own branch and has a name based on the machine it's running from, so there's no conflict between subscriptions or repos even if someone is running the same test locally at the same time as the official run. And the only variables that I reassign during a test are locals within the scope of the test (unless you see some that I missed), so that should be ok.

Reply:
The problem is this class is stateful.  If only is running while the other is in the middle of "SetupParameters", it's going to crash in a horrible fire (especially if it's in the parts where it's in the middle of filling one of the lists).

Any field that has any internal state means the tests can't be runnin in parallel.  Fields/BaseClasses are dangerous in tests (they are dangerous in general, but especially in tests).

Reply:
Good point. Let me thing about it for a bit since some of these tests take 5+ minutes to run (due to Arcade implementation constraints) and running them in parallel would save a lot of time. I'll put the attribute on it for the moment so that I can keep moving forward but I'd like to come up with a better long term solution (it might just be moving again from fields at the cost of duplicated code, I'll need to look at it).

Reply:
Oh right, the other thing that I was running into with refactoring to make the TestParameters into a local for each case is the clean up - right now, as best I understand it, the TearDown is what ensures that the clean up happens correctly if the test case throws an exception. If there's a way to make sure that's handled properly then everything can be an instance specific to that test.

Reply:
`using var parameters = new TestParameters();` will always call Dispose for a local

Reply:
Sorry, missed this comment apparently. That works even if there's an exception thrown? I thought part of the whole reason that test cases had so much structure in general was to handle clean up after exceptions.

Reply:
That's the glory of using.  It will clean it up no matter what when you get to the end up of the using block (or, if you did the statement using, the end of the block containing the using).  I'm not sure why there is so much structure to the tests.

It's a little magic, but Dispose is so common that it's an acceptable level of magic.

Reply:
Whoa... Well I'm impressed. I wasn't clear that it would handle things quite to that level. In that case, locals for all! :)

Reply:
I am afraid that enabling this parallel execution would require considerable refactoring.
I'd approach it like this:
```
using var state = new GivenGitHubFlowWithAllCheckMergePolicy();
await WaitForMergedPullRequestAsync(state);
...
```
The GivenGitHubFlowWithAllCheckMergePolicy() and related Given{Something} methods would serve code reuse purpose (plus readability)

Reply:
I don't quite understand how that's different from moving the definition and initialization for TestParameters and any other non-readonly values into the test case. Shouldn't making everything be local ensure a new instance of the EndToEndLogic, MaestroTestBase (since it's a base for EndToEndLogic, not GitHubFlowTests) and TestParameters classes is created for each test?

Reply:
What was the resolution here?